release/cloud-build-configs/kfserving-controller.cloud-build.yaml

```diff
@@ -0,0 +1,7 @@
+steps:
+- name: 'gcr.io/cloud-builders/go'
+  args: ['install', './cmd/manager']
+  env:  ['PROJECT_ROOT=github.com/kubeflow/kfserving']
+- name: 'gcr.io/cloud-builders/docker'
+  args: ['build', '-t', 'gcr.io/kfserving/kfserving-controller:latest', '.']
+images: ['gcr.io/kfserving/kfserving-controller:latest']
```

Comment:
Is this configuration file the input of Google Cloud Build or else?
I just cannot understand what `images` field means:  dose it mean that multiple steps of `gcr.io/cloud-builders/docker` can build multiple images, only those included by `images` will be pushed into registry?

Reply:
I have a Google Cloud Build Trigger that points to this repository and specifically this file (you should be able to go check it out if you click the link in the readme).

Images is a google cloud build concept for what gets pushed after the build completes. We're doing the most simple cloud build you can do:

1. Build a binary
2. Package it in docker
3. Push the docker image to a repo

Reply:
I am wondering what's the difference between google cloud build and 
https://github.com/kubeflow/kfserving/blob/master/test/scripts/build-kfserving.sh#L40, are they the same thing?

Reply:
They are the same thing. The difference is that one is triggered and the other is pushed.

Build Triggers are a fairly new feature, so I think that's why they were originally not used.

I'm a big fan of trigger-based CI/CD.